python/AMI_NOT_PUBLIC_CHECK/AMI_NOT_PUBLIC_CHECK_test.py

```diff
@@ -0,0 +1,216 @@
+import sys
+import unittest
+try:
+    from unittest.mock import MagicMock, patch, ANY
+except ImportError:
+    import mock
+    from mock import MagicMock, patch, ANY
+import botocore
+from botocore.exceptions import ClientError
+
+##############
+# Parameters #
+##############
+
+# Define the default resource to report to Config Rules
+DEFAULT_RESOURCE_TYPE = 'AWS::::Account'
+
+#############
+# Main Code #
+#############
+
+config_client_mock = MagicMock()
+sts_client_mock = MagicMock()
+ec2_client_mock = MagicMock()
+
+class Boto3Mock():
+    def client(self, client_name, *args, **kwargs):
+        if client_name == 'config':
+            return config_client_mock
+        elif client_name == 'sts':
+            return sts_client_mock
+        elif client_name == 'ec2':
+            return ec2_client_mock
+        else:
+            raise Exception("Attempting to create an unknown client")
+
+sys.modules['boto3'] = Boto3Mock()
+
+rule = __import__('AMI_NOT_PUBLIC_CHECK')
+
+# Checks for scenario wherein no non-compliant resources are present
+class CompliantResourcesTest(unittest.TestCase):
+    lambda_event = {}
+
+    def setUp(self):
+        self.lambda_event = build_lambda_scheduled_event()
+        pass
+
+    def test_scenario_1_compliant_resources(self):
+        describe_images_result = {
+                    'Images': [],
+                    'ResponseMetadata': {'HTTPStatusCode': 200}
+        }
+        ec2_client_mock.describe_images = MagicMock(return_value=describe_images_result)
+        lambda_result = rule.lambda_handler(self.lambda_event, {})
+        expected_response = [
+                	build_expected_response(
+                    		compliance_type='NOT_APPLICABLE',
```

Comment:
The name of the variable input, is not necessary here.

Reply:
removed